### ISSUE TITLE
Azure OpenAI Settings

### ISSUE THREAD

**Issue by jefflavallee** [jekalmin/extended_openai_conversation#231]
More detailed instructions on how to set this up with Azure OpenAI would be helpful.  When I try using my custom endpoint I'm getting the following error:
`Error code: 404 - {'error': {'code': 'DeploymentNotFound', 'message': 'The API deployment for this resource does not exist. If you created the deployment within the last 5 minutes, please wait a moment and try again.'}}`

**Comment by vcdx71** [comment]
Make sure you enabled the Azure OpenAI network setting for All Networks, and you have the correct baseurl from your deployment.

**Comment by itispip** [comment]
> More detailed instructions on how to set this up with Azure OpenAI would be helpful. When I try using my custom endpoint I'm getting the following error: `Error code: 404 - {'error': {'code': 'DeploymentNotFound', 'message': 'The API deployment for this resource does not exist. If you created the deployment within the last 5 minutes, please wait a moment and try again.'}}`

This maybe because you openned "use Tool" swith. If you close it, this error won't happen. But I suspect then you won't able to use those extended function as well which are OpenAI specific.

**Comment by ToosoftwareUser** [comment]
Hi to All,i have the same problem with my azure openai configuration.At this moment i have settings usage for all networks but i receive same error: Sorry, I had a problem talking to OpenAI: Error code: 404 - {'error': {'code': '404', 'message': 'Resource not found'}}. How i can resolve it?

**Comment by bbence84** [comment]
I had the same problem (error), which I have fixed by changing the config. By default it was using the gpt4o-mini model but I haven't setup a deployment with that name. Check your Azure deployments and use one that has been configured.

**Comment by bbence84** [comment]
I have enterered "2024-02-15-preview". 
ALso, the base URL should be based on your Azure setup, it's displayed on the AI Foundry site on the Home page (Azure OpenAI Endpoint)

**Comment by PhotoAtomic** [comment]
looks like it is hard to find the way to configure using azure, i hade luck with this settings
go in the AzureAi Foundry panel
create a deployment

![Image](https://github.com/user-attachments/assets/f3d40293-ac76-4c28-805f-54a9dddf72e2)

from the lefr side menu return in HOME and press "View JSON"

![Image](https://github.com/user-attachments/assets/5b6e0ae0-9cfa-4642-b87e-721f3d566827)

your base uri is at the line "OpenAI Language Model Instance API"

use this as base uri

again from HOME copy the API key 1, this will be your Api key

then in Home Assistant

for the Name: i've used the same name of the deployment i've created (gpt-4o-mini for me)
 Api Key, and Base Uri as told before

For the Api Version, as adviced in the previous post : 2024-02-15-preview


For Organization the name of my subscription, but i don't think it is relevant

**Comment by FoliniC** [comment]
Just to add my 2 cents....
To the base url it's added "/chat/completions"
So I tried to insert "https://YourName-somedata-eastus2.cognitiveservices.azure.com/openai/deployments/gpt-4o"

Too bad that it dind't add the "?api-version=2024-08-01-preview" at the end of the endpoint.
The correct endpoint (as taken from azure AI interface) should be
https://YourName-somedata-eastus2.cognitiveservices.azure.com/openai/deployments/gpt-4o/chat/completions?api-version=2024-08-01-preview

@jekalmin could you please log the URL to make it easier to troubleshoot?
Also, only guessing a solution, is it possible to use the base url "as is" if it contains "azure.com"?

Anyway thanks for the great work!

**Comment by jekalmin** [comment]
Does setting like below help?

```yaml
logger:
  logs:
    openai: debug
```

<img width="986" alt="Image" src="https://github.com/user-attachments/assets/7e5902db-3d35-4cca-a3d5-1fc19dd3c9a2" />

**Comment by FoliniC** [comment]
yes!
https://myname-code-eastus2.cognitiveservices.azure.com/openai/deployments/gpt-4o/chat/completions?api-version=2024-08-01-preview&--//chat/completions "401 Unauthorized"
I workarounded the url adding &--// at the end to "remove" the wrong part.
Don't know if this could be the problem !
I use extended open ai for openAI API without problems and Azure AI with "[Azure OpenAI Conversation Custom Integration](https://github.com/joselcaguilar/azure-openai-ha)"

**Comment by jekalmin** [comment]
I just tried Azure OpenAI and also had `Resource not found` error, and it turned out that `api_version` was not correct.
As [PhotoAtomic](https://github.com/jekalmin/extended_openai_conversation/issues/231#issuecomment-2599142357) mentioned, I created deployment of a model from `Azure AI Foundry` panel and got `api_version` from deployed model URL (`2024-08-01-preview` in my case)


If you want to test in python, try code below first.

```python
import logging
from openai import AsyncAzureOpenAI
import asyncio

logging.basicConfig(level="DEBUG")

client = AsyncAzureOpenAI(
    api_key="your-api-key",
    azure_endpoint="https://your-endpoint",
    api_version="2024-08-01-preview"
)

async def main():
    response = await client.models.list()
    print(response)


if __name__ ==  '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(main())

```

**Comment by FoliniC** [comment]
My fault!!!! 
It works...

Using your python script with data taken from
https://ai.azure.com/resource/playground?wsid=/subscriptions/<subscription guid>
Using "View code" (red arrovw)

![Image](https://github.com/user-attachments/assets/59623d5a-ab02-4a67-a307-4a0afd147897)

There select tab "Key authentication" (red arrow)
then 
api_key="<84 char string>",
    azure_endpoint="https://myname-code-eastus2.openai.azure.com/",
    api_version="<api key>"
as the following screenshot 

![Image](https://github.com/user-attachments/assets/91fc07d6-a9c9-4b67-8d9b-275d098df9c8)

One question: after adding the service is it possible to change api_key, azure_enpoint and api_version?

**Comment by jekalmin** [comment]
No, if you want to change whatever you entered in adding it, you have to remove and re-create it using api_key, azure_endpoint, and api_version.

**Comment by FoliniC** [comment]
ok, no interface to modify... doing some trials a faster way, at my own risk and peril, I can modify modify entry.data in the JSON file .storage/core.config_entries :D

**Comment by jekalmin** [comment]
Can you try [2.0.0-beta3](https://github.com/jekalmin/extended_openai_conversation/releases/tag/2.0.0-beta3) version?
I've added a select box of `API Provider` in which you can choose `Azure OpenAI`

**Comment by jekalmin** [comment]
Released in [2.0.0](https://github.com/jekalmin/extended_openai_conversation/releases/tag/2.0.0)
There's a select box to set "API Provider" to manually set azure.
Domain doesn't have to match `AZURE_DOMAIN_PATTERN` anymore.

<img width="300" height="739" alt="Image" src="https://github.com/user-attachments/assets/ac6114ea-c04d-43d7-83f2-6ffe636a50d9" />
